core/src/main/java/com/datastax/oss/driver/api/core/cql/StatementBuilder.java

```diff
@@ -91,6 +91,7 @@ protected StatementBuilder(StatementT template) {
   @NonNull
   public SelfT setExecutionProfileName(@Nullable String executionProfileName) {
     this.executionProfileName = executionProfileName;
+    this.executionProfile = null;
     return self;
   }
```

Comment:
This solved a bug when combining DAO-level profile and `@StatementAttributes`, but I think it also makes sense for general usage of statement builders.